commit 3e7819886281e077e82006fe4804b0d6b0f5643b
Merge: 3a56e2417329 3c1743a685b1
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Mon Jul 15 14:20:22 2024 -0700

    Merge tag 'for-6.11/block-20240710' of git://git.kernel.dk/linux
    
    Pull block updates from Jens Axboe:
    
     - NVMe updates via Keith:
         - Device initialization memory leak fixes (Keith)
         - More constants defined (Weiwen)
         - Target debugfs support (Hannes)
         - PCIe subsystem reset enhancements (Keith)
         - Queue-depth multipath policy (Redhat and PureStorage)
         - Implement get_unique_id (Christoph)
         - Authentication error fixes (Gaosheng)
    
     - MD updates via Song
         - sync_action fix and refactoring (Yu Kuai)
         - Various small fixes (Christoph Hellwig, Li Nan, and Ofir Gal, Yu
           Kuai, Benjamin Marzinski, Christophe JAILLET, Yang Li)
    
     - Fix loop detach/open race (Gulam)
    
     - Fix lower control limit for blk-throttle (Yu)
    
     - Add module descriptions to various drivers (Jeff)
    
     - Add support for atomic writes for block devices, and statx reporting
       for same. Includes SCSI and NVMe (John, Prasad, Alan)
    
     - Add IO priority information to block trace points (Dongliang)
    
     - Various zone improvements and tweaks (Damien)
    
     - mq-deadline tag reservation improvements (Bart)
    
     - Ignore direct reclaim swap writes in writeback throttling (Baokun)
    
     - Block integrity improvements and fixes (Anuj)
    
     - Add basic support for rust based block drivers. Has a dummy null_blk
       variant for now (Andreas)
    
     - Series converting driver settings to queue limits, and cleanups and
       fixes related to that (Christoph)
    
     - Cleanup for poking too deeply into the bvec internals, in preparation
       for DMA mapping API changes (Christoph)
    
     - Various minor tweaks and fixes (Jiapeng, John, Kanchan, Mikulas,
       Ming, Zhu, Damien, Christophe, Chaitanya)
    
    * tag 'for-6.11/block-20240710' of git://git.kernel.dk/linux: (206 commits)
      floppy: add missing MODULE_DESCRIPTION() macro
      loop: add missing MODULE_DESCRIPTION() macro
      ublk_drv: add missing MODULE_DESCRIPTION() macro
      xen/blkback: add missing MODULE_DESCRIPTION() macro
      block/rnbd: Constify struct kobj_type
      block: take offset into account in blk_bvec_map_sg again
      block: fix get_max_segment_size() warning
      loop: Don't bother validating blocksize
      virtio_blk: Don't bother validating blocksize
      null_blk: Don't bother validating blocksize
      block: Validate logical block size in blk_validate_limits()
      virtio_blk: Fix default logical block size fallback
      nvmet-auth: fix nvmet_auth hash error handling
      nvme: implement ->get_unique_id
      block: pass a phys_addr_t to get_max_segment_size
      block: add a bvec_phys helper
      blk-lib: check for kill signal in ioctl BLKZEROOUT
      block: limit the Write Zeroes to manually writing zeroes fallback
      block: refacto blkdev_issue_zeroout
      block: move read-only and supported checks into (__)blkdev_issue_zeroout
      ...

diff --cc include/linux/blkdev.h
index 24c36929920b,dce4a6bf7307..b8196e219ac2
--- a/include/linux/blkdev.h
+++ b/include/linux/blkdev.h
@@@ -1563,7 -1617,8 +1617,7 @@@ int sync_blockdev(struct block_device *
  int sync_blockdev_range(struct block_device *bdev, loff_t lstart, loff_t lend);
  int sync_blockdev_nowait(struct block_device *bdev);
  void sync_bdevs(bool wait);
- void bdev_statx_dioalign(struct inode *inode, struct kstat *stat);
 -void bdev_statx(struct inode *backing_inode, struct kstat *stat,
 -		u32 request_mask);
++void bdev_statx(struct path *, struct kstat *, u32);
  void printk_all_partitions(void);
  int __init early_lookup_bdev(const char *pathname, dev_t *dev);
  #else
@@@ -1581,7 -1636,8 +1635,8 @@@ static inline int sync_blockdev_nowait(
  static inline void sync_bdevs(bool wait)
  {
  }
- static inline void bdev_statx_dioalign(struct inode *inode, struct kstat *stat)
 -static inline void bdev_statx(struct inode *backing_inode, struct kstat *stat,
++static inline void bdev_statx(struct path *path, struct kstat *stat,
+ 				u32 request_mask)
  {
  }
  static inline void printk_all_partitions(void)